sdk/go/pulumi/printf.go

```diff
@@ -0,0 +1,36 @@
+package pulumi
+
+import (
+	"fmt"
+	"io"
+)
+
+func allArgs(args []interface{}) ArrayOutput {
```

Comment:
This is a bit of a wart that I'd like to fix up, but am happy to do so after the fact.

You would hope that `All(Any(args))` or `ToOutput(Any(args))` would work, but neither does. The implementation of these functions stops recursing and awaiting outputs when it hits a value that does not implement `Input`. In this case, that means we don't look through the `[]interface{}` to find outputs. Normally we stop at non-Input values because there isn't much we can do if we encounter an output--awaiting it likely won't produce a value we can assign to the location due to type differences (the location will be typed as an Output, and the awaited value will not)--but we should recurse through compound values looking for `interface{}` fields.